pkg/components/metallb/template.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metallb has code related to deployment of metallb component.
+package metallb
+
+const chartValuesTmpl = `
+controller:
+  enabled: true
+  {{- if .ControllerNodeSelectors }}
+  nodeSelector:
+  {{- range $key, $value := .ControllerNodeSelectors }}
+    {{ $key }}: "{{ $value }}"
+  {{- end }}
+  {{- end }}
+  {{- if .ControllerTolerationsJSON }}
+  tolerations: {{ .ControllerTolerationsJSON }}
+  {{- end }}
+speaker:
+  enabled: true
+  {{- if .SpeakerNodeSelectors }}
+  nodeSelector:
+  {{- range $key, $value := .SpeakerNodeSelectors }}
```

Comment:
Maybe that would be simpler?
```suggestion
  {{- with .SpeakerNodeSelectors }}
  nodeSelector:
  {{- range $key, $value := . }}
```